financial-qa-search/QAReranker/__init__.py

```diff
@@ -0,0 +1,90 @@
+            - a path or url to a `tensorflow index checkpoint file` (e.g. `./tf_model/model.ckpt.index`). In this case, ``from_tf`` should be set to True and a configuration object should be provided as ``config`` argument.
+            This loading path is slower than converting the TensorFlow
+            checkpoint in a PyTorch model using the provided conversion scripts and loading the PyTorch model afterwards.
+        :param model_name: the path where the model is stored
+        :param max_length: the max length to truncate the tokenized sequences to.
+        """
+
+        super().__init__(*args, **kwargs)
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.model_path = model_path
+        self.max_length = max_length
+
+    def post_init(self):
+        super().post_init()
+        self.device = torch.device("cpu")
+        self.tokenizer = BertTokenizer.from_pretrained(self.pretrained_model_name_or_path, do_lower_case=True)
+        self.model = BertForSequenceClassification.from_pretrained(self.pretrained_model_name_or_path, cache_dir=None, num_labels=2)
+        self.model.load_state_dict(torch.load(self.model_path, map_location=self.device), strict=False)
+        self.model.to(self.device)
+        self.model.eval()
+
+    def get_score(self, query, answer):
+
+        # Create inputs for the model
+        encoded_seq = self.tokenizer.encode_plus(query, answer,
+                                            max_length=self.max_length,
+                                            pad_to_max_length=True,
+                                            return_token_type_ids=True,
+                                            return_attention_mask=True)
+        # Numericalized, padded, clipped seq with special tokens
+        input_ids = torch.tensor([encoded_seq['input_ids']]).to(self.device)
+        # Specify question seq and answer seq
+        token_type_ids = torch.tensor([encoded_seq['token_type_ids']]).to(self.device)
+        # Specify which position is part of the seq which is padded
+        att_mask = torch.tensor([encoded_seq['attention_mask']]).to(self.device)
+        # Don't calculate gradients
+        with torch.no_grad():
+            # Forward pass, calculate logit predictions for each QA pair
+            outputs = self.model(input_ids, token_type_ids=token_type_ids, attention_mask=att_mask)
+        # Get the predictions
+        logits = outputs[0]
+        # Apply activation function
+        rel_score = softmax(logits, dim=1)
+        rel_score = rel_score.numpy()
+        # Probability that the QA pair is relevant
+        rel_score = rel_score[:, 1][0]
+
+        return rel_score
+
+    def score(
+            self, query_meta: Dict, old_match_scores: Dict, match_meta: Dict
+    ) -> "np.ndarray":
+
+        new_scores = [
+            (
+                match_id,
+                self.get_score(query_meta['text'], match_meta[match_id]['text']),
```

Comment:
We should think on a way to batch this into the model. Is your model ready and tested on batches?